### PR TITLE
Fix for multiple goroutines appending to slice, Solves #613

### PR DESCRIPTION
**Pull request by sselph**

### CODE REVIEW

File: src/changes.go [hunk 1]
```diff
@@ -262,6 +262,7 @@ type changeSliceArg struct {
 	localParent, remoteParent string
 	dirList                   []*dirList
 	changeListPtr             *[]*Change
+	mtx                       *sync.Mutex
```
Comment (odeke-em) [inline src/changes.go]: Could you make this instead a non-pointer value and drop the extra the indirection ie
```go
mtx sync.Mutex
```

Comment (odeke-em) [inline src/changes.go]: Also could we call it `mu sync.Mutex`?

File: src/changes.go [hunk 2]
```diff
@@ -473,7 +477,9 @@ func (g *Commands) changeSlice(cslArg *changeSliceArg) {
 
 		childChanges, childClashes, cErr := g.resolveChangeListRecv(clr)
 		if cErr == nil {
+			mtx.Lock()
```
Comment (odeke-em) [inline src/changes.go]: cslArg.mu.Lock()

**Comment by odeke-em** [comment]
Just posted a few comments otherwise LGTM! Thank you.

File: src/changes.go (hunk 1 above)
Comment (sselph) [inline src/changes.go]: changing to mu is fine. I'll send an update for that.

The non-pointer I think will be an issue. Each go routine needs to be using the same mutex to effectively communicate. So if I pass the mutex by value, I don't think they will be. I did a quick test and -race was still showing the race condition.

The other option would be to pass a channel and have the workers return values via the channel and add them to the list using a single routine.

Comment (odeke-em) [inline src/changes.go]: Ahhh, gotcha. Aye aye, go for it. Thank you for investigating this.

Comment (odeke-em) [inline src/changes.go]: I meant to say, you can keep the pointer version of it.

**Comment by sselph** [comment]
No problem it was fun getting to play with the race detector.

**Comment by odeke-em** [comment]
LGTM! Thank you @sselph for this fix.

**Pull request merged**